glommio/src/channels/channel_mesh.rs

```diff
@@ -0,0 +1,312 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the
+// MIT/Apache-2.0 License, at your convenience
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2020 Datadog, Inc.
+//
+use std::cell::Cell;
+use std::fmt::{self, Debug, Formatter};
+use std::sync::{Arc, RwLock};
+
+use futures::future;
+
+use crate::channels::local_channel::{self, *};
+use crate::channels::shared_channel::{self, *};
+use crate::{Local, Result};
+
+#[derive(Debug)]
+pub enum Sender<T: Send + Copy> {
+    Local(LocalSender<T>),
```

Comment:
Why do we need a `LocalSender`? I supposed I will have understood by the time I am don reading, but it would be nice to have a description of the channel architecture for new readers anyway.

The description should include: the role of the local sender, how do messages flow, how do responses return, etc.